src/__experimental__/components/switch/switch.style.js

```diff
@@ -37,7 +38,20 @@ const StyledSwitch = styled.div`
       margin-bottom: 8px;
       margin-right: 100%;
 
-      ${labelWidth && `
+      ${({ hasError, hasWarning, hasInfo }) => css`
+        ${() => hasInfo && css`color: ${theme.colors.info};`}
```

Comment:
I guess it can be just
```
${hasInfo && css`color: ${theme.colors.info};`}
${hasWarning && css`color: ${theme.colors.warning};`}
${hasError && css`color: ${theme.colors.error};`}
```
We don't need additional abstract layer here

Reply:
This might become redundant. Waiting on PR #2252.

Reply:
Updated in commit 26f9d5007